### DISCUSSION TITLE
Writing a permission rule without using collection fields (e.g. based on $CURRENT_USER or any dynamic variable)

### DISCUSSION THREAD

**Discussion by 8byr0** [directus/directus#9569] [Feature Requests]
Hey there!
From what I've seen so far this seems to be not possible yet using role permission rules, tell me if I'm wrong.

It would be useful to have ability to create a rule checking for a value depending on current user(or role or any dynamic variable) like so:
```jsonc
// If user has more than 1 credits, allow
// A hook after action may be used to decrease this value 
{
	"$CURRENT_USER": {
		"credits": {
			"_gte": 1
		}
	}
}

// Another example: Prevent 2 users to share the same role
// This one is totally random, may never happen IRL but who knows?
{
	"$CURRENT_ROLE": {
		"users": {
			"_ncontains": "another-user-id"
		}
	}
}
```

I know it's possible to manage it in a `items.<action>.before` hook but having all rules in the same place (single source of truth) instead of dispatched between hooks and role permissions seems better to make more the system more easy to maintain IMHO.

Would it make sense?

**Comment by 8byr0** [comment]
Is it what your PR handles @licitdev ?https://github.com/directus/directus/pull/9572

**Reply by licitdev** [reply]
Nope, it isn't. Nice idea! 💡
Allows for the fine-tuning of permissions~

**Reply by 8byr0** [reply]
Sure ! 
Well I've no experience w/ vue but I'll have a detailed look to the way it's managed and see if a PR could be worth it. I have other plans for today so I'll wait feedback from @benhaynes or @rijkvanzanten

**Reply by benhaynes** [reply]
If I'm not misunderstanding, I belive this is a sponsored feature that is in the works now. Right @rijkvanzanten ?

**Reply by rijkvanzanten** [reply]
@benhaynes No, that's "the other way":

```json
{
  "field": {
    "_in": "$CURRENT_ROLE.users"
  }
}
```

instead of 

```json
{
  "$CURRENT_ROLE": {
    "name": {
      "_eq": "Admin"
    }
  }
}
```

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
